src/flatfile.cpp

```diff
@@ -72,3 +72,22 @@ size_t FlatFileSeq::Allocate(const CDiskBlockPos& pos, size_t add_size, bool& ou
     }
     return 0;
 }
+
+bool FlatFileSeq::Flush(const CDiskBlockPos& pos, bool finalize)
+{
+    FILE* file = Open(FlatFilePos(pos.nFile, 0)); // Avoid fseek to nPos
+    if (!file) {
+        return error("%s: failed to open file %d", __func__, pos.nFile);
+    }
+    if (finalize && !TruncateFile(file, pos.nPos)) {
+        fclose(file);
+        return error("%s: failed to truncate file %d", __func__, pos.nFile);
+    }
+    if (!FileCommit(file)) {
```

Comment:
In commit "validation: Refactor file flush logic into FlatFileSeq." (5bc3ece663177235606e4ee272ff0de2bac1306a)

Probably ok, but previous code would still call `FileCommit` when `TruncateFile` failed, and now it is skipped.